feature/topic/impl/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/topic/impl/navigation/TopicEntryProvider.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.feature.topic.impl.navigation
+
+import androidx.compose.material3.adaptive.ExperimentalMaterial3AdaptiveApi
+import androidx.compose.material3.adaptive.layout.AnimatedPaneScope
+import androidx.compose.material3.adaptive.navigation3.ListDetailSceneStrategy
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation3.runtime.EntryProviderBuilder
+import androidx.navigation3.runtime.entry
+import com.google.samples.apps.nowinandroid.core.navigation.NiaBackStack
+import com.google.samples.apps.nowinandroid.feature.topic.api.navigation.TopicRoute
+import com.google.samples.apps.nowinandroid.feature.topic.api.navigation.navigateToTopic
+import com.google.samples.apps.nowinandroid.feature.topic.api.TopicScreen
+import com.google.samples.apps.nowinandroid.feature.topic.api.TopicViewModel
+import com.google.samples.apps.nowinandroid.feature.topic.api.TopicViewModel.Factory
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ActivityComponent
+import dagger.multibindings.IntoSet
+
+@Module
+@InstallIn(ActivityComponent::class)
+object TopicModule {
+
+    @OptIn(ExperimentalMaterial3AdaptiveApi::class)
+    @Provides
+    @IntoSet
+    fun provideTopicEntryProviderBuilder(
+        backStack: NiaBackStack,
+    ): EntryProviderBuilder<Any>.() -> @JvmSuppressWildcards Unit = {
```

Comment:
remove `@Suppress`

Reply:
Weird, at one point Hilt was throwing without this. Seems fine now without it. Removed.